### PR TITLE
fix: cant reach the last block in nesting

### PR DESCRIPTION
**Pull request by fundon**
## Before
If we want to move A after B, but not a child of B. The current move can only be a child of B, if B is the last block.

Screenshot 2023-01-12 at 4 30 29 PM

## After

May not be very good in user experience

https://user-images.githubusercontent.com/27926/212127387-83f54f4e-44f7-4b52-8aa8-9f4f68f2b6a2.mov

### CODE REVIEW

File: packages/blocks/src/page-block/default/utils.ts [hunk 1]
```diff
@@ -178,8 +181,30 @@ function binarySearchBlockEditingState(
...
+
+                return {
+                  index: mid,
+                  position: result.blockRect,
+                  model: result.block,
+                };
+              }
+            }
+          } else {
+            if (x < detectRect.left || x > detectRect.left + detectRect.width) {
```
Comment (himself65) [inline packages/blocks/src/page-block/default/utils.ts:205]: use `isPointIn`?

File: packages/blocks/src/page-block/default/utils.ts [hunk 2]
```diff
@@ -178,8 +181,30 @@ function binarySearchBlockEditingState(
         assertExists(blockRect);
 
         if (!options?.skipX) {
-          if (x < detectRect.left || x > detectRect.left + detectRect.width) {
-            return null;
+          if (dragging) {
+            if (block.depth) {
+              let depth = Math.floor((blockRect.left - x) / 26);
```
Comment (himself65) [inline packages/blocks/src/page-block/default/utils.ts:186]: what is 26?

Comment (fundon) [reply packages/blocks/src/page-block/default/utils.ts:186]: padding-left

File: packages/blocks/src/page-block/default/utils.ts (hunk 1 above)
Comment (fundon) [reply packages/blocks/src/page-block/default/utils.ts:205]: Ok, add it back

File: packages/blocks/src/page-block/default/utils.ts (hunk 2 above)
Comment (himself65) [reply packages/blocks/src/page-block/default/utils.ts:186]: I think we should put everything related into a constant

**Pull request merged**